sql-plugin/src/main/spark311/scala/org/apache/spark/sql/hive/rapids/shims/GpuSaveAsHiveFile.scala

```diff
@@ -0,0 +1,89 @@
+/*
+ * Copyright (c) 2023, NVIDIA CORPORATION.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/*** spark-rapids-shim-json-lines
+{"spark": "311"}
+{"spark": "312"}
+{"spark": "313"}
+{"spark": "320"}
+{"spark": "321"}
+{"spark": "321cdh"}
+{"spark": "321db"}
+{"spark": "322"}
+{"spark": "323"}
+{"spark": "330"}
+{"spark": "330cdh"}
+{"spark": "330db"}
+{"spark": "331"}
+{"spark": "332"}
+{"spark": "333"}
+{"spark": "340"}
```

Comment:
Great code dedupe!

If this is indeed what it looks like, tagged for all  shim builds, we should move it to the out of spark311 to src/main/scala and drop `shims` from the package name.

```bash
ls -ald sql-plugin/src/main/spark3* | wc -l
16
```

We should add a check to shimplify to fail the build with the suggestion to unshim code that became universal across existing shims due to refactoring and/or dropping shims. Similar to the orphaned code message.

Reply:
Can we create a follow-on issue for Shimplify?

Reply:
I created #8158 as soon as I was done with my original comment :)